packages/expo-print/android/src/main/java/expo/modules/print/PrintModule.kt

```diff
@@ -0,0 +1,227 @@
+          }
+        }
+        promise.resolve(Bundle().apply {
+          putString("uri", uri)
+          putInt("numberOfPages", numberOfPages)
+          if (base64 != null) putString("base64", base64)
+        })
+      }
+
+      override fun onRenderError(errorCode: String?, errorMessage: String?, exception: Exception?) {
+        promise.reject(errorCode, errorMessage, exception)
+      }
+    })
+  }
+
+  private fun printDocumentToPrinter(document: PrintDocumentAdapter, options: Map<String?, Any?>) {
+    val printManager = moduleRegistry
+      .getModule(ActivityProvider::class.java)
+      .currentActivity
+      ?.getSystemService(Context.PRINT_SERVICE) as? PrintManager
+    val attributes = getAttributesFromOptions(options)
+    printManager?.print(jobName, document, attributes.build())
+  }
+
+  private fun getAttributesFromOptions(options: Map<String?, Any?>): PrintAttributes.Builder {
+    val orientation = if (options.containsKey("orientation")) options["orientation"] as String? else null
+    val builder = PrintAttributes.Builder()
+
+    // @tsapeta: Unfortunately these attributes might be ignored on some devices or Android versions,
+    // in other words it might not change the default orientation in the print dialog,
+    // however the user can change it there.
+    if ((ORIENTATION_LANDSCAPE == orientation)) {
+      builder.setMediaSize(PrintAttributes.MediaSize.UNKNOWN_LANDSCAPE)
+    } else {
+      builder.setMediaSize(PrintAttributes.MediaSize.UNKNOWN_PORTRAIT)
+    }
+
+    // @tsapeta: It should just copy the document without adding extra margins,
+    // document's margins can be controlled by @page block in CSS.
+    builder.setMinMargins(PrintAttributes.Margins.NO_MARGINS)
+    return builder
+  }
+
+  @Throws(IOException::class)
+  private fun generateFilePath(): String {
+    return FileUtils.generateOutputPath(context.cacheDir, "Print", ".pdf")
+  }
+
+  @Throws(IOException::class)
+  private fun encodeFromFile(file: File): String {
+    val randomAccessFile = RandomAccessFile(file, "r")
+    val fileBytes = ByteArray(randomAccessFile.length().toInt())
+    randomAccessFile.readFully(fileBytes)
+    return Base64.encodeToString(fileBytes, Base64.NO_WRAP)
+  }
+
+  private fun decodeDataURI(uri: String): InputStream {
+    val base64Index = uri.indexOf(";base64,")
+    val plainBase64 = uri.substring(base64Index + 8)
+    val byteArray = Base64.decode(plainBase64, Base64.DEFAULT)
+    return ByteArrayInputStream(byteArray)
+  }
+
+  @Throws(IOException::class)
+  private fun loadAndClose(destination: ParcelFileDescriptor, callback: WriteResultCallback, input: InputStream) {
+    val output = FileOutputStream(destination.fileDescriptor)
+    val buf = ByteArray(1024)
+    var bytesRead: Int
+    while ((input.read(buf).also { bytesRead = it }) > 0) {
+      output.write(buf, 0, bytesRead)
+    }
+    callback.onWriteFinished(arrayOf(PageRange.ALL_PAGES))
+    try {
+      input.close()
+      output.close()
```

Comment:
Instead of close, you can use `use` function from kotlin library -> https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html

Reply:
I changed it for OutputStream only, because InputStream is creating in another function.